File: packages/age-signals/android/src/main/java/io/capawesome/capacitorjs/plugins/agesignals/classes/results/RequestAgeRangeResult.java
package io.capawesome.capacitorjs.plugins.agesignals.classes.results;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.agesignals.enums.AgeRangeStatus;
import io.capawesome.capacitorjs.plugins.agesignals.interfaces.Result;

public class RequestAgeRangeResult implements Result {

    @NonNull
    private final AgeRangeStatus status;

    @Nullable
    private final GetAgeRangeResult ageRangeResult;

    public RequestAgeRangeResult(@NonNull AgeRangeStatus status, @Nullable GetAgeRangeResult ageRangeResult) {
        this.status = status;
        this.ageRangeResult = ageRangeResult;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = ageRangeResult == null ? new JSObject() : ageRangeResult.toJSObject();
        result.put("status", status.name());
        return result;
    }
}

File: packages/age-signals/ios/Plugin/Classes/Results/RequestAgeRangeResult.swift
import Capacitor

@objc class RequestAgeRangeResult: NSObject, Result {
    private let ageRange: AgeRange?
    private let status: AgeRangeStatus

    init(status: AgeRangeStatus, ageRange: AgeRange? = nil) {
        self.status = status
        self.ageRange = ageRange
    }

    func toJSObject() -> AnyObject {
        var result = JSObject()
        result["ageRange"] = ageRange?.toJSObject()
        result["status"] = status.rawValue
        return result as AnyObject
    }
}
